### PR TITLE
drivers: sbs_gauge: Add support for Alarm properties

### PR DESCRIPTION
**Pull request by madhurimaparuchuri**
RemainingCapacityAlarm(r/w) and RemainingTimeAlarm(r/w)

### CODE REVIEW

File: include/zephyr/drivers/fuel_gauge.h [hunk 1]
```diff
@@ -84,6 +84,10 @@ extern "C" {
 #define FUEL_GAUGE_SBS_ATRATE_TIME_TO_EMPTY FUEL_GAUGE_SBS_ATRATE_TIME_TO_FULL + 1
 /** AtRateOK (boolean) */
 #define FUEL_GAUGE_SBS_ATRATE_OK	    FUEL_GAUGE_SBS_ATRATE_TIME_TO_EMPTY + 1
+/** Remaining Capacity Alarm (mAh or 10mWh) */
+#define FUEL_GAUGE_REMAINING_CAPACITY_ALARM FUEL_GAUGE_SBS_ATRATE_OK + 1
```
Comment (aaronemassey) [inline include/zephyr/drivers/fuel_gauge.h:88]: Please add `SBS` prefix as this is particularly specific to SBS

File: tests/drivers/fuel_gauge/sbs_gauge/src/test_sbs_gauge.c [hunk 2]
```diff
@@ -134,23 +134,66 @@ ZTEST_USER_F(sbs_gauge_new_api, test_set_some_props_failed_returns_failed_prop_c
...
 {
 	uint16_t word = BIT(15) | BIT(0);
-	struct fuel_gauge_set_property set_prop = {
-		/* Valid property */
-		.property_type = FUEL_GAUGE_SBS_MFR_ACCESS,
-		/* Set Manufacturer's Access to 16 bit word*/
-		.value.sbs_mfr_access_word = word,
+	struct fuel_gauge_set_property set_props[] = {
+		{
+			/* Valid property */
```
Comment (aaronemassey) [inline tests/drivers/fuel_gauge/sbs_gauge/src/test_sbs_gauge.c:139]: Was this a `clang-format` change?

**Review by aaronemassey — CHANGES_REQUESTED** [review]

File: tests/drivers/fuel_gauge/sbs_gauge/src/test_sbs_gauge.c (hunk 2 above)
Comment (madhurimaparuchuri) [reply tests/drivers/fuel_gauge/sbs_gauge/src/test_sbs_gauge.c:139]: No actually, earlier it's single property, now it's list of properties, hence the indentation is added

File: include/zephyr/drivers/fuel_gauge.h (hunk 1 above)
Comment (madhurimaparuchuri) [reply include/zephyr/drivers/fuel_gauge.h]: Refactored it in new revision

**Review by aaronemassey — APPROVED** [review]

**Review by teburd — APPROVED** [review]
I think this is ok, though I'd look hard in a second round whether some of these can be consolidated into common fuel gauge properties. The max17055 for example has alerts available for battery removal, insertion, under/over voltage, under/over current, under/over soc with settable values.

Do the SBS alarms not match up with those in some way, especially with regards to the capacity?

**Comment by aaronemassey** [comment]
> I think this is ok, though I'd look hard in a second round whether some of these can be consolidated into common fuel gauge properties. The max17055 for example has alerts available for battery removal, insertion, under/over voltage, under/over current, under/over soc with settable values.
> 
> Do the SBS alarms not match up with those in some way, especially with regards to the capacity?

That's a good point. I err on the side of caution to *not* put out generic properties.

@madhurimaparuchuri said she's going to look into this next week and follow up.

**Pull request merged**